open-grant-proposals/Filecoin Financing Plan.md

```diff
@@ -0,0 +1,122 @@
+hardware and mining pool can publish financing plans to the entire network. 
+Filecoin financing plans will include the total financing amount, the timeline, the usage plan, repayment amount (yield indicator) and repayment plan(installment). Anyone can make an investment by transferring FIL to the financing plan address, and financing repayment will be returned to the transfer account. 
+To secure the benefit of the investors, the financing plan itself shows investors the credit information of the financier. At the same time, the logic of freezing financing fund and income automatic repayment of the financing plan will be implemented in the lotus-miner, and to freeze financing fund will guarantee that the financing fund can only be used to increase the mining pool power. The Automatic income repayment will guarantee that the investors' income is not subject to the individual intentions of the financier. 
+
+## Value
+
+Please describe in more detail why this proposal is valuable for the Filecoin ecosystem. Answer the following questions:
+- What are the benefits to getting this right?
+A: The project can spread the pressure of high cost to join Filecoin, to reduce the risk for individuals to participate, to lower the cost threshold for Filecoin participation, and to attract more users to get involved.
+- What are the risks if you don't get it right?
+A: If we didn't get it right, the investors may lost their capital.
+- What are the risks that will make executing on this project difficult?
+A: The credit rating of financiers is not comprehansive or accurate enough, and that makes investors afraid to invest. 
+
+## Deliverables
+
+1. deliver the lotus-miner implementation code for financing plan
+2. deliver the web browing portal (financing plan explorer) code for financing plan and access services.
+
+## Technical Breakdown
+
+Filecoin financing plan detailed process:
+1. Add two control address types for miner, which means on the basis of the worker/owner address, we add the financing address and the repayment freeze address.
+2. A sealing sector first determines whether the financing address has some balance, If the address has, the sector will use the balance; if not, the sealing sector will follow the original logic to process.
+3. A financing plan includes features below:
+Financing address
+Financing start time
+Financing end time
+Financing total amount (At the financing end time, if financing total amount has not yet met, the financing plan will be canceled automatically.)
+Repayment total amount
+Repayment start time
+Repayment end time
+Repayment period
+Repayment period option (3 options: month, week, day)
+4. During the repayment time, the miner balance is frozen to pay for the current repayment. When the added-up balance meets the repayment limit, the balance will be automatically transferred to the repayment freeze address.
```

Comment:
Adding more complexity to the miner actor's token holding logic is not currently an option.

Reply:
If we add any logic it should only be at withdraw